lib/init/grass.py

```diff
@@ -264,6 +265,43 @@ def wxpath(*args):
     return os.path.join(_WXPYTHON_BASE, *args)
 
 
+def count_wide_chars(s):
+    """Returns the number of wide CJK characters in a string.
+
+    :param str s: string
+    """
+    return sum(unicodedata.east_asian_width(c) in 'WF' for c in
+            (s if sys.version_info.major >= 3 else unicode(s)))
+
+
+def f(fmt, *args):
+    """Adjusts fixed-width string specifiers for wide CJK characters and
+    returns a formatted string. Does not support named arguments yet.
+
+    :param str fmt: format string
+    :param *args: arguments for the format string
+    """
+    matches = []
+    # https://docs.python.org/3/library/stdtypes.html#old-string-formatting
+    for m in re.finditer('%([#0 +-]*)([0-9]*)(\.[0-9]*)?([hlL]?[diouxXeEfFgGcrsa%])', fmt):
```

Comment:
This differs from line 28 in your patch of lib/gis/aprintf.c - just spotted and likely on purpose (I don't exactly know what it does...).

Reply:
Good catch. Looks like Python doesn't support all the C specifiers?